_ap/18/index.adoc

```diff
@@ -3,76 +3,198 @@ num: 18
+
+==== Compatibility matrix
+
+Kiota https://github.com/microsoft/kiota/issues/2020[compatibility matrix] covers the current Red Hat use-cases, specifically:
+
+* Java
+* Go
+* Python
+* TypeScript
+
+The generator doesn't need to stitch or even know about specific frameworks and/or libraries as they rely only on interfaces (called “abstractions”), and implementation libraries are swappable.
+
+==== External Owners (Microsoft)
+
+The project is maintained and owned by Microsoft, specifically by the team working on Microsoft Graph.
+We will be having some direct discussions with them soon around their plans for the project and a possible collaboration.
+Initial impressions indicate that they are very receptive to improvements and collaborations.
+But, of course, they will always own the project in the same way that Red Hat owns our first-party OSS projects.
+
+On the flip side, this is a possible occasion for a strong collaboration between two of the major players in the Software industry that can drive more and more community adoption of the technology/stack.
+
+Using OpenAPI Tools we are in a very similar situation except from the fact that it's more a "pure" Open Source project.
+
+==== Legal Requirements (Terms Acceptance and Export Control)
+
+Kiota and the relevant component are released under the MIT license.
+
+=== Additional Reading
+
+Kiota design: https://microsoft.github.io/kiota/extending/
+Some libraries are already used [in Red Hat projects](https://github.com/microsoft/kiota-abstractions-go/network/dependents).
+
+=== Threat Model
+
+image::threat-model.png[]
+
+* Given that Kiota is going to produce source code and runs on the client side there are no specific threats for RH SaaS
+* A Code Generator could produce malicious or corrupted generated code.
+* Clients with large numbers of dependencies could be vulnerable if not refreshed
+* A swapped implementation may not be well supported
+
+=== Alternatives Considered / Rejected
+
+==== Keep going with the current generator
+
+The DevExp team has identified as a major piece of maintenance the upgrade of the currently outdated https://github.com/OpenAPITools/openapi-generator[openapi-generator].
+The current SDKs are full of quirks and edge cases in most cases hacked around to end up with a working implementation.
+Having to keep the https://docs.google.com/spreadsheets/d/1fwmRmapDdFRU-B7qFBUuXyHMDjzXtRMY-jhbtxDVtyU/edit#gid=0[hacks and workarounds] is a major blocker to updating toward newer versions of the generator.
+
+The monolithic approach to a generation used by openapi-generator also offers challenges in a modular approach to bug fixes and improvements.
+
+==== IBM
+
+We have had extensive discussions with IBM regarding their approach to https://github.com/IBM/ibm-cloud-sdk-common[client SDK] generation and we very much like their governance model over APIs produced in the company.
```

Comment:
I am missing why this was ruled out?

Reply:
The section on IBM was based on confidential information.
I stripped it down to a minimum that can be safely shared.

Which are the options to keep part of this document private?
Should I make an "Addendum" in the form of a Google Doc linked here?
@tombentley do you have any suggestion here?

Reply:
GDoc seems like a good idea to me @andreaTP

Reply:
done: https://github.com/bf2fc6cc711aee1a0c2a/architecture/pull/90/commits/6bf32c97b56f4c73c3b363481f23ef38b8e854b6